imap/ctl_conversationsdb.c

```diff
@@ -501,6 +501,29 @@ static int next_diffable_record(struct cursor *c)
     }
 }
 
+static void printer(const char *type, struct cursor *c)
+{
+    if (!verbose) return;
+    static struct buf keybuf = BUF_INITIALIZER;
+    static struct buf databuf = BUF_INITIALIZER;
+    buf_reset(&keybuf);
+    buf_reset(&databuf);
+    size_t i;
+    for (i = 0; i < c->keylen; i++) {
+        if (c->key[i] > 31 && c->data[i] < 127)
+            buf_putc(&keybuf, c->key[i]);
+        else
+            buf_printf(&keybuf, "<%d>", (int)c->key[i]);
+    }
+    for (i = 0; i < c->datalen; i++) {
+        if (c->data[i] > 31 && c->data[i] < 127)
```

Comment:
This one looks bug-free, but could also probably just use isprint()

Reply:
ooh nice.  It should also be casting the value to (uint8_t) and using %u really.   Bah.  But it's quite readable.